src/NuGet.Core/NuGet.Packaging/Signing/SignatureTrustResult.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Represents a signature check result and any additional information
+    /// needed to display to the user.
+    /// </summary>
+    public class SignatureTrustResult
+    {
+        /// <summary>
+        /// Trust result
+        /// </summary>
+        public SignatureTrust Trust { get; }
```

Comment:
When using `signtool` to verify a signature it gives some information on success or error, would this class be the one that include that?

Signtool includes:
- Signature file path
- Sha1 hash of signature
- Signing cert and chain
- Timestamp
- Timestamper cert and chain
- Errors (if any)
- Warnings (if any)

Reply:
Everything needed to create the error messages would be in this class. What is available is likely going to be based on what the lower level command that the trust provider uses to verify will provide.